### ISSUE TITLE
Better window title appearance

### ISSUE THREAD

**Issue by Keith94** [zydezu/ModernX#49]
Would like to see the window title at the top appear more modern like a YouTube video, i.e. text is bigger with vignette at the top and no black background, and no window controls. We can't seem to disable to window controls separately... `windowcontrols=no` will hide the entire title bar.

Also wasn't obvious at first but I had to use `force-media-title=` in mpv.conf to change the text output, since `title=` wasn't working

![brave_ZeTWUIz7Eb](https://github.com/user-attachments/assets/0ea2f9b6-ed14-4a26-b471-b9f3353ee992)

![mpv_Bv7F9z0MgD](https://github.com/user-attachments/assets/11c94e2f-ac85-4cf7-a082-19bc74a359a6)

**Comment by Samillion** [comment]
> Also wasn't obvious at first but I had to use force-media-title= in mpv.conf to change the text output, since title= wasn't working

I just noticed that. I think there are two things causing this issue:

https://github.com/zydezu/ModernX/blob/142ed71508396fa7300f84647354381118225853/modernx.lua#L1175-L1181

This `mp.set_property("title", mediatitle)` should be moved inside the nested condition for `user_opts.dynamictitle`.

For example:
```lua
	local mediatitle = mp.get_property("media-title")

	if (mp.get_property("filename") ~= mediatitle) and user_opts.dynamictitle then
		mp.set_property("title", mediatitle)
		user_opts.title = "${media-title}"
	end
```

I'm not sure why this is in the end of the file: (I commented it out)
https://github.com/zydezu/ModernX/blob/142ed71508396fa7300f84647354381118225853/modernx.lua#L3996

Will test to see if my changes break anything.

**Comment by Samillion** [comment]
Looking at `checktitle()` function, I'm not sure why `mp.set_property("title", mediatitle)` is added there. This should make `user_opts.dynamictitle` work just fine

```lua
function checktitle()
	local mediatitle = mp.get_property("media-title")

	if (mp.get_property("filename") ~= mediatitle) and user_opts.dynamictitle then
		user_opts.title = "${media-title}"
	end
```

I might be missing something though.

**Comment by Samillion** [comment]
Ok, so after testing out a few things it's more clear the use of the function. Mostly affects web videos (path is url), which is part of the whole dynamic title/description features (basically show media title instead of filename, with added parameters such as artist, album..etc)

The feature is great, but I honestly think it should all operate only and if at least one of the web related `user_opts` is set to true, otherwise it should leave the title property as set in `mpv.conf`/default.

@zydezu If I ever find the time, I might create a PR draft to improve the functionality, if you don't mind. Though it might be a while, as I am traveling soon. :'(

**Comment by zydezu** [comment]
ok.

**Comment by Samillion** [comment]
This thing kinda got out of hand with me lol. I ended up forking the entire project (with credit given to all modern forks).

The basic idea was: re-do it all to match mpv/osc.lua, in terms of standards, functions and so on, to ensure compatibility and functionality in the future.

Feel free to look it over: https://github.com/Samillion/ModernZ

**Comment by zydezu** [comment]
Oh wow… to be honest I’m not that good at lua… bit I will take a look at this when I have some time :P

**Comment by Keith94** [comment]
Fixed in [0.3.9](https://github.com/zydezu/ModernX/releases/tag/0.3.9)
